addons/gamification/models/gamification_karma_tracking.py

```diff
@@ -1,69 +1,131 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-import calendar
-
 from dateutil.relativedelta import relativedelta
 
-from odoo import api, fields, models
+from odoo import api, fields, models, _
+from odoo.tools import date_utils
 
 
 class KarmaTracking(models.Model):
     _name = 'gamification.karma.tracking'
     _description = 'Track Karma Changes'
     _rec_name = 'user_id'
-    _order = 'tracking_date DESC'
+    _order = 'user_id, tracking_date desc, id desc'
+
+    def _get_selection_origin(self):
+        return [('res.users', 'User (Manual change)')]
 
-    user_id = fields.Many2one('res.users', 'User', index=True, readonly=True, required=True, ondelete='cascade')
-    old_value = fields.Integer('Old Karma Value', required=True, readonly=True)
-    new_value = fields.Integer('New Karma Value', required=True, readonly=True)
+    user_id = fields.Many2one('res.users', 'User', index=True, required=True, ondelete='cascade')
+    old_value = fields.Integer('Old Karma Value', compute='_compute_old_value', store=True)
+    new_value = fields.Integer('New Karma Value', required=True)
     consolidated = fields.Boolean('Consolidated')
-    tracking_date = fields.Date(default=fields.Date.context_today)
+
+    tracking_date = fields.Datetime(default=fields.Datetime.now())
+    reason = fields.Text(default=_("Added Manually"), string='Reason')
```

Comment:
The default will not be translated with the user lang, you should do `lambda: _("Added Manually")` I think

Reply:
done